### ISSUE TITLE
uturn incorrectly detected and penalized

### ISSUE THREAD

**Issue by fenwuyaoji** [valhalla/valhalla#4635]
Hi, I'm a freshman of Valhalla, I found a case below:

request parameter: 
  car (auto)
Ideally link: 
  https://valhalla.openstreetmap.de/directions?profile=car&wps=100.65868142013117%2C13.759098685948903%2C100.66893643970043%2C13.75446497964344

description:
  Same as https://github.com/valhalla/valhalla/issues/3886, Valhalla made a wrong route of car mode according to the u-turn detection and penalty.
  Three short ways (964183843, 964183844, 964184744) can make a turn heading to the destination, and all properties of them are the same, except the length. Valhalla detects 964183843 and 964183844 u-turns and adds a cost of 600 to them. This penalty makes 964184744 the best candidate and it looks wired.
My question is: Is there a way to prevent cases like that or improve it? 

  Official result:
<img width="1618" alt="image" src="https://github.com/valhalla/valhalla/assets/35750880/0fdd42d6-af0b-43e5-8fed-e872873966c7">
  Wished result:
<img width="1646" alt="image" src="https://github.com/valhalla/valhalla/assets/35750880/eb8fa955-d0a2-45eb-8d36-3188268c9e4a">

**Comment by nilsnolde** [comment]
Hm I’d say it mad the _wrong_ uturn, but it very much seems a uturn would be needed right? Albeit a much “earlier” one as you pointed out. Not sure why that isn’t happening.

**Comment by fenwuyaoji** [comment]
已收到您的邮件，我将及时查看并回复，谢谢                                                                                                                     王鑫

**Comment by kevinkreiser** [comment]
yeah detecting what is and isn't a utrun is trickier than you think. we have another open issue about this which I made a suggestion to fix this using updates to the edge label in the issue you linked https://github.com/valhalla/valhalla/issues/3886#issuecomment-1413085733

**Comment by fenwuyaoji** [comment]
@kevinkreiser Yes, yes, I understand it's a tough challenge, and I know we can't fix all bad cases. Appreciate your great work and your response, let me figure out whether we can do some work to help this issue.

**Comment by fenwuyaoji** [comment]
How about we set it as a param in cost_options? so that we can adjust it more dynamically.
